inc/saibuffer.h

```diff
@@ -484,21 +484,29 @@ typedef enum _sai_buffer_profile_attr_t
      * @brief Dynamic threshold for the shared usage
      *
      * The threshold is set to the 2^n of available buffer of the pool.
-     * Mandatory when SAI_BUFFER_POOL_THRESHOLD_MODE = SAI_BUFFER_THRESHOLD_MODE_DYNAMIC
```

Comment:
Buffer profile threshold mode has 3 values
Inherit buffer pool mode, which is the default value, means look at buffer pool which the profile is attached to (through SAI_BUFFER_PROFILE_ATTR_POOL_ID), and inherit from that pool threshold mode (SAI_BUFFER_POOL_ATTR_THRESHOLD_MODE)
There are 2 explicit modes - SAI_BUFFER_PROFILE_THRESHOLD_MODE_STATIC, SAI_BUFFER_PROFILE_THRESHOLD_MODE_DYNAMIC, that basically set explicit buffer profile mode

Once the mode is decided according to this logic, if the mode is static only static attributes are relevant (SAI_BUFFER_PROFILE_ATTR_SHARED_STATIC_TH). If it's dynamic, only dynamic attributes are relevant (SAI_BUFFER_PROFILE_ATTR_SHARED_DYNAMIC_TH)

This logic hasn't changed

In the original buffer proposal we had just the 2 explicit buffer profile threshold modes
Several months ago Dell added the concept of inherit from pool
This pull request doesn't change anything in the logic IMO, just fixes the metadata